stdlib/public/SDK/Foundation/TypePreservingNSNumber.mm

```diff
@@ -178,16 +181,17 @@ - (Class)classForCoder {
 DEFINE_INIT(float, Float)
 DEFINE_INIT(double, Double)
 DEFINE_INIT(CGFloat, CGFloat)
-DEFINE_INIT(bool, Bool)
 
 #undef DEFINE_INIT
 
 SWIFT_CC(swift) extern "C" uint32_t
 _swift_Foundation_TypePreservingNSNumberGetKind(
   NSNumber *NS_RELEASES_ARGUMENT _Nonnull self_) {
   uint32_t result = 0;
-  if ([self_ isKindOfClass: [_SwiftTypePreservingNSNumber class]]) {
+  if ([self_ isKindOfClass:[_SwiftTypePreservingNSNumber class]]) {
     result = ((_SwiftTypePreservingNSNumber *) self_)->tag;
+  } else if (CFGetTypeID(self_) == CFBooleanGetTypeID()) {
```

Comment:
`assert(strcmp(@YES.objCType, @encode(BOOL)) == 0);` 

where as CFGetTypeID of a subclass of NSNumber may not return `CFBooleanGetTypeID()` unless `_cfTypeID` is overridden.